File: code/go/ch17x/src/luago/compiler/parser/parser.go
package parser

import . "luago/compiler/ast"
import . "luago/compiler/lexer"

/* recursive descent parser */

func Parse(chunk, chunkName string) *Block {
	lexer := NewLexer(chunk, chunkName)
	block := parseBlock(lexer)
	lexer.NextTokenOfKind(TOKEN_EOF)
	return block
}

File: code/java/ch17/src/main/java/com/github/zxh0/luago/compiler/parser/Parser.java
package com.github.zxh0.luago.compiler.parser;

import com.github.zxh0.luago.compiler.ast.Block;
import com.github.zxh0.luago.compiler.lexer.Lexer;
import com.github.zxh0.luago.compiler.lexer.TokenKind;

public class Parser {

    public static Block parse(String chunk, String chunkName) {
        Lexer lexer = new Lexer(chunk, chunkName);
        Block block = BlockParser.parseBlock(lexer);
        lexer.nextTokenOfKind(TokenKind.TOKEN_EOF);
        return block;
    }

}
